### PR TITLE
Add 5 Animation/generating figures

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add 5. "Animation/generating figures"

 

### Details

- Show how to animate/generating figures using the PyVista API to make awesome online documentation. (10 min for talk, 10 min for exercise)

### CODE REVIEW

**Comment by tkoyama010** [comment]
This is a draft, but I would be grateful for a review.

**Comment by tkoyama010** [comment]
I am counting the number of words by following the script (ref: https://stackoverflow.com/questions/71194571/word-count-of-markdown-cells-in-jupyter-notebook). 
```python
import json

with open("5_animation_generating_figures.ipynb") as json_file:
    data = json.load(json_file)

wordCount = 0
for each in data["cells"]:
    cellType = each["cell_type"]
    if cellType == "markdown":
        content = each["source"]
        for line in content:
            temp = [
                word for word in line.split() if "#" not in word
            ]  # we might need to filter for more markdown keywords here
            wordCount = wordCount + len(temp)

print(wordCount)
```
Now the number of words is 87. [There are 1,500 words in a 10-minute speech.
](https://wordcounter.net/blog/2016/04/04/10948_how-many-words-10-minute-speech.html).

We can add more content.

**Comment by tkoyama010** [comment]
I gathered the example *.ipynb files and run the following script.
```python
import json

wordCount = 0
for file_name in [
   "5_animation_generating_figures.ipynb",
   "flying_edges.ipynb",
   "gif.ipynb",
   "linked.ipynb",
   "movie.ipynb",
   "moving_cmap.ipynb",
   "orbit.ipynb",
]:
    with open(file_name) as json_file:
        data = json.load(json_file)
    for each in data["cells"]:
        cellType = each["cell_type"]
        if cellType == "markdown":
            content = each["source"]
            for line in content:
                temp = [
                    word for word in line.split() if "#" not in word
                ]  # we might need to filter for more markdown keywords here
                wordCount = wordCount + len(temp)

print(wordCount)
```
Now it is 283 words and we can add more content.

**Comment by tkoyama010** [comment]
It is not only animation but also generating so we can also add the filter's contents.

**Comment by tkoyama010** [comment]
I gathered again the  *.ipynb files and run the following script.
```python
import json

wordCount = 0
for file_name in [
   "5_animation_generating_figures.ipynb",
   "background_image.ipynb",
   "cmap.ipynb",
   "flying_edges.ipynb",
   "gif.ipynb",
   "linked.ipynb",
   "movie.ipynb",
   "moving_cmap.ipynb",
   "orbit.ipynb",
   "scalar-bars.ipynb",
   "screenshot.ipynb",
   "texture.ipynb",
   "themes.ipynb",
   "vector-component.ipynb",
   "volume.ipynb",
]:
    with open(file_name) as json_file:
        data = json.load(json_file)
    for each in data["cells"]:
        cellType = each["cell_type"]
        if cellType == "markdown":
            content = each["source"]
            for line in content:
                temp = [
                    word for word in line.split() if "#" not in word
                ]  # we might need to filter for more markdown keywords here
                wordCount = wordCount + len(temp)

print(wordCount)
```
Now it is 1968 words and it is enough as contents.

**Comment by tkoyama010** [comment]
Now I ran.
```python
nbmerge 5_animation_generating_figures.ipynb \
   background_image.ipynb \
   cmap.ipynb \
   flying_edges.ipynb \
   gif.ipynb \
   linked.ipynb \
   movie.ipynb \
   moving_cmap.ipynb \
   orbit.ipynb \
   scalar-bars.ipynb \
   screenshot.ipynb \
   texture.ipynb \
   themes.ipynb \
   vector-component.ipynb \
   volume.ipynb > merged.ipynb

mv merged.ipynb 5_animation_generating_figures.ipynb
```

**Comment by tkoyama010** [comment]
Some exercises are needed.

**Comment by tkoyama010** [comment]
We can switch 1 or 2 examples to exercises.

**Comment by tkoyama010** [comment]
- [ ] Check the link of methods and functions.

**Comment by tkoyama010** [comment]
- [ ] Add the title and the abstract of the section.

**Comment by banesullivan** [comment]
@tkoyama010 and @akaszynski, thinking I'm wrapped up with this section. Would appreciate a review if either of you have time. Please note that the exercises are taken from PyVista's gallery and only mildly modified (so hopefully don't need much review)

File: tutorial/05_figures/README.rst [hunk 1]
```diff
@@ -1,17 +1,273 @@
...
+* Label points in 3D space along side your meshes
+* Creating side-by-side comparisons
+* Making a dataset transparent or using a scalar value to map opacity
+* Adding textures/images draped over a mesh (texture mapping)
+* Use sophisticated lighting techniques like smooth shading or Eye Dome Lighting
+* Creating animations as GIFs or movie files
+
+This section will overview PyVista's :class:`pyvista.Plotter` API and how to perform a these tasks.
+The goal of this lesson is not to be a comprehensive overview of PyVista's plotting API, but
+rather to demonstrate how it works and how you can learn to use it!
```
Comment (tkoyama010) [inline tutorial/05_figures/README.rst:28]:
```suggestion
This section will overview PyVista's :class:`pyvista.Plotter` API and how to perform these tasks.
The goal of this lesson is not to be a comprehensive overview of PyVista's plotting API, but
rather to demonstrate how it works and how you can learn to use it!
```

File: tutorial/05_figures/README.rst [hunk 2]
```diff
@@ -1,17 +1,273 @@
...
+
+    p.subplot(0, 0)
+    p.add_mesh(pv.Sphere())
+
+    p.subplot(0, 1)
+    p.add_mesh(pv.Cube())
+
+    p.show()
+
+Below is an example of a side-by-side comparisons of the contours and slices of a single datasets.
```
Comment (tkoyama010) [inline tutorial/05_figures/README.rst:160]:
```suggestion
Below is an example of side-by-side comparisons of the contours and slices of a single dataset.
```

**Comment by banesullivan** [comment]
Thanks, @tkoyama010!

**Pull request merged**